plugins/readtimer.koplugin/main.lua

```diff
@@ -66,26 +125,109 @@ function ReadTimer:remaining()
     end
 end
 
-function ReadTimer:remainingTime()
+-- can round
+function ReadTimer:remainingTime(round)
     if self:scheduled() then
         local remainder = self:remaining()
+        if round then
+            if round < 0 then -- round down
+                remainder = remainder - 59
+            elseif round == 0 then
+                remainder = remainder + 30
+            else -- round up
+                remainder = remainder + 59
+            end
+            remainder = math.floor(remainder * (1/60)) * 60
+        end
+
         local hours = math.floor(remainder * (1/3600))
         local minutes = math.floor(remainder % 3600 * (1/60))
         local seconds = math.floor(remainder % 60)
         return hours, minutes, seconds
     end
 end
 
+function ReadTimer:addAdditionalHeaderContent()
+    self.ui.crelistener:addAdditionalHeaderContent(self.additional_header_content_func)
+    self.update_status_bars(self, -1)
+end
+function ReadTimer:addAdditionalFooterContent()
+    self.ui.view.footer:addAdditionalFooterContent(self.additional_footer_content_func)
+    self.update_status_bars(self, -1)
+end
+
+function ReadTimer:removeAdditionalHeaderContent()
+    self.ui.crelistener:removeAdditionalHeaderContent(self.additional_header_content_func)
+    self.update_status_bars(self, -1)
+    UIManager:broadcastEvent(Event:new("UpdateHeader"))
+end
+function ReadTimer:removeAdditionalFooterContent()
+    self.ui.view.footer:removeAdditionalFooterContent(self.additional_footer_content_func)
+    self.update_status_bars(self, -1)
```

Comment:
When you are calling it directly (that is: not passing it to scheduleIn), you can call it `self:update_status_bars(-1)` - as you indeed define it as a classic method.
It's the scheduleIn usage which is non-classic, and we will understand self needs to be passed - no need to make the odd usage visible here where it does not need to be odd.